FrontEnd/FrontEnd.Web/Components/Pages/Recipes/SingleRecipe.razor

```diff
@@ -0,0 +1,102 @@
+        <FluentSkeleton Height="2em" Width="100%"/>
+        <FluentSkeleton Height="2em" Width="100%"/>
+    </FluentStack>
+}
+else if (_recipe == null)
+{
+    <FluentLabel Typo="Typography.H1" MarginBlock="10px">Recipe not found.</FluentLabel>
+    <FluentLabel Typo="Typography.Body" MarginBlock="10px">The requested recipe doesn't exist.</FluentLabel>
+}
+else
+{
+    <div style="height: 14rem; overflow: hidden; display: flex; justify-content: center; align-items: center;">
+        <img src="@_recipeUrl" alt="@_recipe.Title"/>
+    </div>
+    <FluentStack Style="margin-block: 10px;">
+        <FluentLabel Typo="Typography.H1">@_recipe.Title</FluentLabel>
+        <FluentSpacer/>
+        <FluentAnchor Href="" Appearance="Appearance.Accent">Edit</FluentAnchor>
+        <FluentAnchor Href="" Appearance="Appearance.Accent">Delete</FluentAnchor>
+    </FluentStack>
+    <FluentGrid AdaptiveRendering>
+        <FluentGridItem sm="12" md="6">
+            <FluentStack Orientation="Orientation.Vertical" Style="min-width: 500px;">
+                <FluentOverflow>
+                    <FluentOverflowItem>
+                        <FluentBadge Appearance="Appearance.Accent">
+                            <FluentIcon Color="Color.Disabled" Value=@(new Icons.Regular.Size16.Timer())/>@_recipeCookingTime
+                        </FluentBadge>
+                    </FluentOverflowItem>
+                    <FluentOverflowItem>
+                        <FluentBadge Appearance="Appearance.Accent">
+                            <FluentIcon Color="Color.Disabled" Value=@(new Icons.Regular.Size16.PersonBoard())/>@_recipeServings
+                        </FluentBadge>
+                    </FluentOverflowItem>
+                    @foreach (var item in _tagNames)
+                    {
+                        <FluentOverflowItem>
+                            <FluentBadge Appearance="Appearance.Accent">@item</FluentBadge>
+                        </FluentOverflowItem>
+                    }
+                </FluentOverflow>
+                <p>@_description</p>
+                <FluentLabel Color="Color.Accent" Typo="Typography.H4">Ingredients</FluentLabel>
+                <FluentDataGrid Items=_ingredients.AsQueryable() Style="width: 100%;" ShowHover GenerateHeader="GenerateHeaderOption.None">
+                    <PropertyColumn Property="@(ingredient => ingredient)" />
+                </FluentDataGrid>
+            </FluentStack>
+        </FluentGridItem>
+        <FluentGridItem sm="12" md="6">
+            <FluentLabel Color="Color.Accent" Typo="Typography.H4">Instructions</FluentLabel>
+            <ol>
+                @foreach (var instruction in _recipe.Instructions)
+                {
+                    <li style="margin-block: 10px;">@instruction</li>
+                }
+            </ol>
+        </FluentGridItem>
+    </FluentGrid>
+}
+
+@code {
+    [Parameter]
+    public int Id { get; set; }
```

Comment:
NIT: probably init instead of set

Reply:
You're right, though this will be overrided by the prerequisite.